packages/next/src/server/load-components.ts

```diff
@@ -46,7 +46,7 @@ export type LoadComponentsReturnType = {
   getStaticProps?: GetStaticProps
   getStaticPaths?: GetStaticPaths
   getServerSideProps?: GetServerSideProps
-  ComponentMod: any
+  ComponentMod: NextModule
```

Comment:
The `LoadComponentsReturnType` has been expanded to support a type parameter, which is used by the dreaded `RenderOpts` for both the App Pages and Page render flows.